src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -75,56 +98,118 @@ class SDCPUThread
      * execution of an instruction, this should be freely usable as a tag for
      * our dynamic instructions.
      */
-    typedef uint64_t InstTag;
-    InstTag numInstsCommitted = 0;
-    InstTag numInstsSpeculated = 0;
+    InstSeqNum numInstsCommitted = 0;
 
     /**
-     * The SimpleThread class
+     * The SimpleThread class already contains all internal state necessary to
+     * hold onto the state of a basic CPU, demonstrated by its use in many
+     * other CPU models. Until we find a significant need to hold particular
+     * permanent committed state different from a regular CPU in this library,
+     * there is no reason to redefine the wheel.
+     * 
+     * The committed state represents the state resulting from the complete
+     * correct execution of all instructions up to the point represented by
+     * numInstsCommitted.
      */
     SimpleThread _committedState;
 
     // BEGIN Speculative state
 
     /**
-     * Instruction execution will pass through an ExecContext interface,
-     * therefore in most cases, it will be tied to speculative state, instead
-     * of the "real" state of the system.
+     * We hold onto the InstSeqNum of the last instruction successfully
+     * committed, for ease of checking if older instruction dependencies have
+     * already been satisfied.
      */
-    SDCPUExecContext _isaInterface;
+    InstSeqNum lastCommittedInstNum;
 
+    /**
+     * We hold a "speculative" PC ahead of the "true" committed PC, in order to
+     * track what it is we're fetching next, prior to instruction issue. The
+     * "true" PC is only advanced during instruction commit.
+     */
     TheISA::PCState _speculativePC;
```

Comment:
I think "fetching pc" would be a better term. Or maybe "decoding pc". I think we're going to want to decouple fetch from decode.

For fetch we're working with *addresses*, not a PC. For decode, we're working with a (potentially speculative) PC.